CONTRIBUTING.md

```diff
@@ -52,6 +52,40 @@ If you are maintaining a labwc package for another distro feel free to open an i
 Let's try to stick to sircmpwn's [coding style]. If you're not used to it, you
 can use [checkpatch.pl] to run a few simple formatting checks.
 
+Specific nuances that apply to labwc:
+
+- We prefer `sizeof(*foo)` to `sizeof(struct foo)`
+
+### The use of glib
+
+We try to keep the use of glib pretty minimal for the following reasons:
+
+- The use of glib has been known to make AddressSanitiser diagnose false
+  positives and negatives.
+- Log messages coming from glib functions look inconsistent.
+- The use of glib functions, naming-conventions and iterators in a code base
+  that is predominantly ANSI C creates a clash which makes readability and
+  maintainability harder.
+- Mixing gmalloc()/malloc() and respective free()s can create problems with
+  memory pools
+
+Having said that, with our use of cairo and pango we depend on glib-2.0 anyway
+so linking with it and making use of some of its helper functions comes for free
+and can keep the code simpler.
+
+For example, if we were going to carry out extensive string manipulation,
+GString and utf8 helpers would be okay. Some functions such as
+`g_utf8_casefold()` would be pretty hard to write from scratch and are
+fine to use. Having said that, labwc does not do much string-work and thus
+far none of these have been needed.
+
+The following functions are used today and are deemed acceptable by the core
+devs:
+
+- `g_shell_parse_argv()`
+- `g_strsplit()`
+- `g_strcmp0`
```

Comment:
> are deemed acceptable by the core devs:

[...]
> `g_strcmp0`

hrm.. by `most of the core devs`? :p

I would still argue against this specific function but if the project lead decides that that specific functions is fine I am fine with it as well and thus `the core devs` would include myself.